core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -97,6 +97,49 @@ package object config {
     .toSequence
     .createWithDefault(Nil)
 
+  // Blacklist confs
+  private[spark] val BLACKLIST_ENABLED =
+    ConfigBuilder("spark.blacklist.enabled")
+    .booleanConf
+    .createOptional
+
+  private[spark] val MAX_TASK_ATTEMPTS_PER_NODE =
+    ConfigBuilder("spark.blacklist.maxTaskAttemptsPerNode")
+    .intConf
+    .createWithDefault(2)
+
+  private[spark] val MAX_FAILURES_PER_EXEC =
+    ConfigBuilder("spark.blacklist.maxFailedTasksPerExecutor")
+    .intConf
+    .createWithDefault(2)
+
+  private[spark] val MAX_FAILURES_PER_EXEC_STAGE =
+    ConfigBuilder("spark.blacklist.maxFailedTasksPerExecutorStage")
```

Comment:
As discussed in the design doc, I think this would be slightly more clear (and the hierarchy of configurations would be more clear) if this were spark.blacklist.stage.maxFailedTasksPerExecutor